dvc/path/__init__.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+from dvc.utils.compat import urlparse
+
+
+class BasePathInfo(object):
+    scheme = None
+
+    def __init__(self, path, url):
+        self.path = path
+        self.url = url
+
+    def __str__(self):
+        return self.url
+
+
+class SSHPathInfo(BasePathInfo):
+    scheme = "ssh"
+
+    def __init__(self, host, user, port, path, url):
+        super(SSHPathInfo, self).__init__(path, url)
+        self.host = host
+        self.user = user
+        self.port = port
+
+
+class HDFSPathInfo(BasePathInfo):
```

Comment:
Let's split these classes into corresponding files like `dvc/path/hdfs.py`, so it is easier to manage.